src/com/cft/contactmerge/FirstAndLastNamesMatchLogic.java

```diff
@@ -28,13 +73,145 @@ private static AnswerType getMatchingResultFromMatchingMethodsList(String contac
         return AnswerType.no;
     }
 
-    private static boolean isFirstAndLastNamesExactMatchCaseSensitive(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+    /************************
+     * Compare Name Methods *
+     ************************/
+
+    private static boolean isMatchMatchCaseSensitive(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
         return (contactOneFirstName.equals(contactTwoFirstName) && contactOneLastName.equals(contactTwoLastName));
     }
 
-    private static AnswerType isFirstAndLastNamesExactMatchCaseInsensitive(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
-        // fill in matching code here
+    private static boolean isMatchCaseInsensitive(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+        return (contactOneFirstName.equalsIgnoreCase(contactTwoFirstName) && contactOneLastName.equalsIgnoreCase(contactTwoLastName));
+    }
 
-        return AnswerType.no;
+    private static boolean isMatchWithWhiteSpaces(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+        return (contactOneFirstName.trim().equals(contactTwoFirstName.trim()) && contactOneLastName.trim().equals(contactTwoLastName.trim()));
+    }
+
+    private static boolean isMatchWithWhiteSpaces_CaseInsensitive(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+        return (contactOneFirstName.trim().equalsIgnoreCase(contactTwoFirstName.trim()) && contactOneLastName.trim().equalsIgnoreCase(contactTwoLastName.trim()));
+    }
+
+    private static boolean isMatchWithPunctuations(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
```

Comment:
Should be "isMatchWithoutPunctuation()".

Reply:
I'm a little confused by this request... If the method is given a name "with" a punctuation (. , - _ ') in it... wouldn't it be trying to match against a name "with/without" punctuations?
vs
Given all names "without" punctuation to match? (in which case this method would never be called)

Reply:
ah I get what you mean.... if punctuations are removed (names "without" punctuations)... do the names match. Okay I can make that change.